tests/sentry/profiles/test_task.py

```diff
@@ -998,6 +998,6 @@ def test_process_profile_task_should_not_emit_profile_duration_outcome(
     )
 
     if "profiler_id" not in profile:
-        assert _track_outcome.call_count == 1
+        assert _track_outcome_legacy.call_count == 1
     else:
-        assert _track_outcome.call_count == 0
+        assert _track_outcome_legacy.call_count == 0
```

Comment:
Would be good to replicate the tests in this file for the new mode, i.e. assert on `_track_outcome` and verify that it's called with the correct arguments.